src/transport/transport.cc

```diff
@@ -937,8 +938,14 @@ void ConvectionTransport::output_data() {
     output_vector_gather();
     //}
 
+	unsigned int ndofs = dh_->max_elem_dofs();
+	unsigned int idof; // iterate over indices
+	std::vector<LongIdx> indices(ndofs);
     for (unsigned int sbi = 0; sbi < n_substances(); sbi++) {
-    	fill_output_data(out_conc[sbi], output_field_ptr[sbi]);
+        for (auto cell : dh_->own_range()) {
+            cell.get_loc_dof_indices(indices);
+            for(idof=0; idof<ndofs; idof++) output_field_ptr[sbi]->get_data_vec()[ indices[idof] ] = (*out_conc[sbi].data_ptr())[ ndofs*cell.elm_idx()+idof ];
```

Comment:
There should be no such copy. We should have a field for the mobile concentration and get  the  `out_conc` vector or rather VecMPI from it. A lot of code can possibly be simplified using the VecMPI wrapper in the  ConvectionTransport consistently.